### PR TITLE
feat(codemirror): Add HTMLEditor and CSSEditor

### PR DESCRIPTION
**Pull request by jbrundage**
Signed-off-by: Jaman Brundage <jbrundage372@gmail.com>

## Checklist:
- [x] The commit message is properly formatted and free of typos.
  - [x] The commit message title makes sense in a changelog, by itself.
  - [ ] The commit message includes a "fixes" reference if appropriate.
  - [x] The commit is signed.
- [x] The change has been fully tested:
  - [ ] I have viewed all related gallery items
  - [ ] I have viewed all related dermatology items
- [ ] There are no similar instances of the same problem that should be addressed
  - [ ] I have addressed them here
  - [ ] I have raised new issues to address them separately

## Testing:
Tested the two new editors in a stand-alone html page.

### CODE REVIEW

**Comment by jbrundage** [comment]
Pic of HTMLEditor:
image

File: demos/gallery/samples/codemirror/HTMLEditor.js [hunk 1]
```diff
@@ -0,0 +1,7 @@
+import { HTMLEditor } from "@hpcc-js/codemirror";
+
+new HTMLEditor()
+    .target("target")
+    .html(atob("PCFET0NUWVBFIGh0bWw+CjxodG1sPgo8aGVhZD4KICAgIDxtZXRhIGNoYXJzZXQ9InV0Zi04IiAvPgogICAgPG1ldGEgaHR0cC1lcXVpdj0iWC1VQS1Db21wYXRpYmxlIiBjb250ZW50PSJJRT1lZGdlIj4KICAgIDx0aXRsZT5ocGNjLWpzPC90aXRsZT4KICAgIDxtZXRhIG5hbWU9InZpZXdwb3J0IiBjb250ZW50PSJ3aWR0aD1kZXZpY2Utd2lkdGgsIGluaXRpYWwtc2NhbGU9MSI+CiAgICA8c2NyaXB0IHNyYz0icGFja2FnZXMvdXRpbC9kaXN0L2luZGV4LmpzIj48L3NjcmlwdD4KICAgIDxzY3JpcHQgc3JjPSJwYWNrYWdlcy9jb21tb24vZGlzdC9pbmRleC5qcyI+PC9zY3JpcHQ+CiAgICA8c2NyaXB0IHNyYz0icGFja2FnZXMvYXBpL2Rpc3QvaW5kZXguanMiPjwvc2NyaXB0PgogICAgPHNjcmlwdCBzcmM9InBhY2thZ2VzL2NvZGVtaXJyb3IvZGlzdC9pbmRleC5qcyI+PC9zY3JpcHQ+CjwvaGVhZD4KPGJvZHk+CiAgICA8ZGl2IGlkPSJ0YXJnZXQiIHN0eWxlPSJ3aWR0aDo1MDBweDtoZWlnaHQ6NTAwcHg7Ij48L2Rpdj4KICAgIDxzY3JpcHQ+CiAgICAgICAgbmV3IHdpbmRvd1siQGhwY2MtanMvY29kZW1pcnJvciJdLkNTU0VkaXRvcigpCiAgICAgICAgICAgIC50YXJnZXQoInRhcmdldCIpCiAgICAgICAgICAgIC5jc3MoIiIpCiAgICAgICAgICAgIC5yZW5kZXIoKQogICAgICAgICAgICA7CiAgICA8L3NjcmlwdD4KPC9ib2R5Pgo8L2h0bWw+"))
```
Comment (GordonSmith) [inline demos/gallery/samples/codemirror/HTMLEditor.js]: Please expand html to human text for consistency with the other samples.

File: packages/codemirror/src/CSSEditor.ts [hunk 2]
```diff
@@ -0,0 +1,22 @@
+import { Editor } from "./Editor";
+
+export class CSSEditor extends Editor {
+    options(): any {
+        return {
+            ...super.options(),
+            ...{
+                mode: "text/css"
+            }
```
Comment (GordonSmith) [inline packages/codemirror/src/CSSEditor.ts]: Simplify to:
```
        return {
            ...super.options(),
            mode: "text/css"
            }
```

File: packages/codemirror/src/HTMLEditor.ts [hunk 3]
```diff
@@ -0,0 +1,22 @@
+import { Editor } from "./Editor";
+
+export class HTMLEditor extends Editor {
+    options(): any {
+        return {
+            ...super.options(),
+            ...{
+                mode: "htmlmixed"
+            }
```
Comment (GordonSmith) [inline packages/codemirror/src/HTMLEditor.ts]: Simplify to:
```
        return {
            ...super.options(),
            mode: "htmlmixed"
            }
```

**Pull request merged**